### ISSUE TITLE
Merging Hue smart app and device

### ISSUE THREAD

**Issue by unparagoned** [BobRak/OpenHAB-Smartthings#34]
There is a really good default Hue discovery app on Smartthings, which works pretty much perfectly with the openhab hue emulator.  So you can go the other way and import your openhab items into Smartthings. This conflicts with you binding on the Smartthings side, since both devices need to be on the same network address. I kind of found a work around by importing items then moving that device to say MAC/24, updating that smart app, then adding your smart app. 

Everything in the first half an hour or so is very sketchy,  with devices going offline, and commands pilling up for 10 minutes then all going through together in one one way while the other direction is fine. 

So it would be quite useful if you could integrate the discovery of the hue emulator by the smartthings hub into your binding so it's fully integrated system. There is some example older code for basic Hue discovery on the Smartthings site so it might not be that much more work. But I've got limited understanding of how the smartthings side of things work.

**Comment by BobRak** [comment]
Thanks for using my Smartthings binding. I know it doesn't support all devices perfectly. Over time I hope to make it support more devices. I don't currently own any Hue Lighting and that makes it difficult (or maybe impossible) for me to integrate it. I did a little reading about Hue. Are you using the "Philips Hue Ethernet Bridge"? If not, how is the Hue device communicating with the Smartthings hub? Can you provide me links to the "default Hue discovery app" and "openhab hue emulator" you referenced above. Especially the openhab emulator may make it possible for me to develop for it without owning the product.  I would also be interested in knowing what your openHab thing and item definitions look like for the Hue.

Currently I am preparing to move within the next couple of months after having lived at the same place for 24 years. It has been like a full time job. So, I will try to fix any bugs found promptly but enhancements are going to be a lower priority for now.

Thanks,

Bob

**Comment by unparagoned** [comment]
Thanks. Sorry if I wasn't clear. In openhab you can write your items with the Homekit tags so that they look like a "HUE" device. Openhab comes with a HUE emulator by default, so any light you can connected to openhab of any brand can be seen as a HUE device. So my Alexa and Home think they are adding a HUE bridge and devices they are just adding the openhab Hue emulator and my normal bulbs. Similarly for Smartthings. I also have a HUE device but that's not important since I don't need a two way binding with a real HUE bridge. 
https://github.com/openhab/openhab2-addons/tree/master/addons/io/org.openhab.io.hueemulation
https://community.openhab.org/t/new-binding-hue-emulator-amazon-echo-integration/7944

So yeh you don't need to have seen HUE or have one, it's all about the how the emulator is one of the foundations that allows openhab to be so compatible with various systems and devices. But once you've settled in to you new place send me a pm on the openhab community forums and I'll see if I can help you out with a Hue starter pack.

**Comment by unparagoned** [comment]
I also just found out the your app also stops openhab from communicating with Alexa and other devices over the LAN. Like I mentioned openahb uses the hue emulator to talk to alexa, google home, etc. But your app takes control causing problems. I'm pretty lucky in that I just added virtual NIC on my VM and have your app and he hue emulator on different virtual nics so that's not a problem for me but it's likely to be for quite a few people.
